packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -31,86 +34,145 @@ export type AppLayoutSidebarNavProps = {
 						? false
 						: Boolean(group.options?.disableGroupPadding);
 
-					const prevGroup = items[idx - 1];
+					const prevGroup = items[index - 1];
 					const prevGroupDisableGroupPadding = prevGroup
 						? Array.isArray(prevGroup)
 							? false
 							: Boolean(prevGroup.options?.disableGroupPadding)
 						: false;
 
 					return (
-						<Fragment key={idx}>
+						<Fragment key={index}>
 							{!isFirstItem ? (
 								<AppLayoutSidebarNavDivider
 									// As the divider is placed at the start of each group, padding top is determined by the previous group
 									disablePaddingTop={prevGroupDisableGroupPadding}
 									disablePaddingBottom={disableGroupPadding}
 								/>
 							) : null}
-							{groupItems.map((item, idx) => (
-								<AppLayoutSidebarNavListItem
-									activePath={activePath}
-									background={background}
-									item={item}
-									key={idx}
-								/>
-							))}
+							{groupItems.map((item, index) => {
+								const isOpen =
+									subLevelVisible === 'always' ||
+									hasSubLevelActiveItem(item.items, activePath);
+
+								return (
+									<AppLayoutSidebarNavListItem
+										activePath={activePath}
+										background={background}
+										isOpen={isOpen}
+										item={item}
+										key={index}
+									/>
+								);
+							})}
 						</Fragment>
 					);
 				})}
-			</Flex>
-		</Flex>
+			</Stack>
+		</Stack>
 	);
-}
+};
 
 type AppLayoutSidebarNavListItemProps = {
 	activePath?: string;
 	background?: 'body' | 'bodyAlt';
+	isOpen: boolean;
 	item: NavItem;
 };
 
 function AppLayoutSidebarNavListItem({
 	activePath,
 	background,
+	isOpen,
 	item,
 }: AppLayoutSidebarNavListItemProps) {
 	const Link = useLinkComponent();
-	const { endElement, icon: Icon, label, ...restItemProps } = item;
+	const {
+		endElement,
+		icon: Icon,
+		isActive,
+		items,
+		label,
+		level,
+		...restItemProps
+	} = item;
 	const { closeMobileMenu } = useAppLayoutContext();
 
 	// Link list item
 	if ('href' in item) {
-		const active = item.href === activePath;
+		const isCurrentPage = item.href === activePath;
 		return (
 			<AppLayoutSidebarNavItemInner
 				background={background}
 				hasEndElement={Boolean(endElement)}
-				isActive={activePath === item.href}
+				isCurrentPage={isCurrentPage}
+				isActive={isCurrentPage || isOpen}
+				level={item.level}
 				onClick={closeMobileMenu} // Let the click event bubble up and close the menu on press of interactive item
 			>
-				<Link aria-current={active ? 'page' : undefined} {...restItemProps}>
-					{Icon ? <Icon color="inherit" /> : null}
+				<Link
+					aria-current={isCurrentPage ? 'page' : undefined}
+					{...restItemProps}
+				>
+					{Icon && level === 1 && <Icon color="inherit" />}
+					{level === 2 && <span aria-hidden>&ndash;</span>}
 					<span>{label}</span>
 					{endElement}
 				</Link>
+
+				{(isOpen || isCurrentPage) && (
+					<Stack as="ul">
```

Comment:
This should only be rendered if it has `items` otherwise each one gets an empty `ul`
<img width="304" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/b49c9b54-c718-4911-887b-7d9d175c9a6b">

Reply:
- [x] only show `ul` when it has length
- [x] fix conditional styling